fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/properties/RecordLayerPropertyStorage.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * RecordLayerPropertyStorage.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.properties;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.RecordCoreException;
+import com.apple.foundationdb.record.logging.LogMessageKeys;
+import com.google.common.collect.ImmutableMap;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * Adopter can use this mapping of {@link RecordLayerPropertyKey}s to {@link RecordLayerPropertyValue}s to configure Record Layer parameters.
+ * The instances of {@link RecordLayerPropertyKey} and how they are used are defined in Record Layer.
+ * Adopters can populate {@link RecordLayerPropertyValue} and put them into this storage.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class RecordLayerPropertyStorage {
+    private static final RecordLayerPropertyStorage EMPTY_PROPERTY_STORAGE = new RecordLayerPropertyStorage(ImmutableMap.of());
+
+    @Nonnull
+    private final ImmutableMap<RecordLayerPropertyKey<?>, RecordLayerPropertyValue<?>> propertyMap;
```

Comment:
Minor (no need to change), but it would generally be better to just use `Map<RecordLayerPropertyKey<?>, RecordLayerPropertyValue<?>` instead of a concrete implementation.